### ISSUE TITLE
Sign macOS release binaries

### ISSUE THREAD

**Issue by hrls** [Atleta-network/atleta#152]
![Image](https://github.com/user-attachments/assets/01b44797-87e4-4b20-b785-48c744032fb7)

**Comment by hrls** [comment]
Managing certificates is only available for developers enrolled in a developer program or members of an organization’s team in a developer program.

**Comment by ales-tsurko** [comment]
for now we can use self-signed certificates, it's a known story for macOS users to reopen an app via the security settings.

**Comment by hrls** [comment]
Does a self-signed certificate bypass the need to allow in settings to run binary from the release archive?
If not, then from a user experience perspective it is the same as an unsigned one .

**Comment by hrls** [comment]
It was also found that the binary from the release archive does not have the _executable_ flag.
Yes, it's as simple as `chmod +x …` but involves extra command.

**Comment by ales-tsurko** [comment]
> Does a self-signed certificate bypass the need to allow in settings to run binary from the release archive? If not, then from a user experience perspective it is the same as an unsigned one .

Yes, you should run it via the security settings. You can't run unsigned binary at all.

**Comment by hrls** [comment]
> You can't run unsigned binary at all

Actually you can. It's almost the same flow but with some another hidden button named _Allow anyway …_ or so.

**Comment by ales-tsurko** [comment]
Ok, code signing is very confusing. Looks like there are two things: ad-hoc signing and notarization. You can't run a binary without both at all, but you can run it via the settings in case of ad-hoc signing without the notarization. And when it's built on macOS ARM, it's ad-hoc by default. Issues happen in case of cross-compilation, and we had one similar recently until we self-signed the binary.

**Comment by hrls** [comment]
https://docs.github.com/en/actions/use-cases-and-examples/deploying/installing-an-apple-certificate-on-macos-runners-for-xcode-development
